In [1]:
import pandas as pd

A continuacion realizo la lectura de los csv

In [2]:
penetracion_p100hab = pd.read_csv('./DatasetsObligatorios/Penetración-poblacion.csv')
penetracion_pcia = pd.read_csv('./DatasetsObligatorios/Penetracion-hogares.csv')
accesos_baf = pd.read_csv('./DatasetsObligatorios/Totales Dial-BAf.csv')
accesos_baf_pcia = pd.read_csv('./DatasetsObligatorios/Dial-BAf.csv')
accesos_tec = pd.read_csv('./DatasetsObligatorios/Accesos Por Tecnología.csv')
accesos_vmd = pd.read_csv('./DatasetsObligatorios/Totales VMD.csv')
accesos_vel_prov = pd.read_csv('./DatasetsObligatorios/Accesos por velocidad.csv')
accesos_totales = pd.read_csv('./DatasetsObligatorios/Totales Accesos por velocidad.csv')
accesos_prov_vel = pd.read_csv('./DatasetsObligatorios/Velocidad_sin_Rangos.csv')
ingresos = pd.read_csv('./DatasetsObligatorios/Ingresos .csv')
accesos_loc = pd.read_csv('./DatasetsObligatorios/Acc_vel_loc_sinrangos.csv')
acceso_tec_loc = pd.read_csv('./DatasetsObligatorios/Accesos_tecnologia_localidad.csv')
mapa_conect = pd.read_csv('./DatasetsObligatorios/mapa_conectividad.csv')
metadata = pd.read_csv('./DatasetsComplementarios/Metadata_mundial.csv')
mundial = pd.read_csv('./DatasetsComplementarios/mundial.csv')


---

Comienzo a analizar los diferentes Datos de los csv para realizar un proceso de ETL, para su posterior EDA, y su final presentacion en un Dashboard.
* El objetivo de este analisis profundo es minimizar la cantidad de datasets y poseer los datos correctos y necesarios para su presentacion final en PowerBi.

In [3]:
penetracion_p100hab.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 4 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Año                       984 non-null    int64  
 1   Trimestre                 984 non-null    int64  
 2   Provincia                 984 non-null    object 
 3   Accesos por cada 100 hab  984 non-null    float64
dtypes: float64(1), int64(2), object(1)
memory usage: 30.9+ KB


In [4]:
penetracion_pcia.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 4 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Año                           984 non-null    int64  
 1   Trimestre                     984 non-null    int64  
 2   Provincia                     984 non-null    object 
 3   Accesos por cada 100 hogares  984 non-null    float64
dtypes: float64(1), int64(2), object(1)
memory usage: 30.9+ KB


In [5]:
#Combino columnas año, Trimestre, Provincia
df1_comb = pd.merge(penetracion_p100hab, penetracion_pcia, on =['Año', 'Trimestre','Provincia'])
df1_comb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 5 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Año                           984 non-null    int64  
 1   Trimestre                     984 non-null    int64  
 2   Provincia                     984 non-null    object 
 3   Accesos por cada 100 hab      984 non-null    float64
 4   Accesos por cada 100 hogares  984 non-null    float64
dtypes: float64(2), int64(2), object(1)
memory usage: 38.6+ KB


In [6]:
df1_comb.sample(2)

Año  Trimestre   Provincia  Accesos por cada 100 hab  \
701  2016          4     Córdoba                 17.571291   
654  2017          2  Corrientes                  8.265166   

     Accesos por cada 100 hogares  
701                     57.696719  
654                     31.386121

---------------

In [7]:
accesos_tec.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 986 entries, 0 to 985
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Año           985 non-null    object 
 1   Trimestre     985 non-null    object 
 2   Provincia     984 non-null    object 
 3   ADSL          984 non-null    float64
 4   Cablemodem    984 non-null    float64
 5   Fibra óptica  984 non-null    float64
 6   Wireless      984 non-null    float64
 7   Otros         984 non-null    float64
 8   Total         984 non-null    float64
dtypes: float64(6), object(3)
memory usage: 69.5+ KB


In [8]:
accesos_tec.head(5)

Año Trimestre        Provincia      ADSL  Cablemodem  Fibra óptica  \
0  2024         1     Buenos Aires  203297.0   2790663.0     1799013.0   
1  2024         1  Capital Federal   70108.0   1177475.0      203756.0   
2  2024         1        Catamarca    5901.0     11025.0       56542.0   
3  2024         1            Chaco   11343.0     58929.0       64463.0   
4  2024         1           Chubut   26306.0     77217.0       26356.0   

   Wireless     Otros      Total  
0  150290.0  101294.0  5044557.0  
1    4493.0   48879.0  1504711.0  
2    1384.0      81.0    74933.0  
3    8404.0   12404.0   155543.0  
4   30867.0    9930.0   170676.0

In [9]:
accesos_tec.isna().sum()

Año             1
Trimestre       1
Provincia       2
ADSL            2
Cablemodem      2
Fibra óptica    2
Wireless        2
Otros           2
Total           2
dtype: int64

In [10]:
accesos_tec = accesos_tec.dropna()

In [11]:
#visualizo el Dataframe y encuentro '*' en Año y Trimestre. Reemplazo el *
accesos_tec = accesos_tec.copy()
accesos_tec['Año'].unique() 
accesos_tec['Año'] = accesos_tec['Año'].str.replace('[^0-9]', '').str.rstrip('*')
accesos_tec['Año'] = accesos_tec['Año'].astype(int)
#access_tec.head(5)
accesos_tec['Año'].unique() 

array([2024, 2023, 2022, 2021, 2020, 2019, 2018, 2017, 2016, 2015, 2014])

In [12]:
accesos_tec.info()

<class 'pandas.core.frame.DataFrame'>
Index: 984 entries, 0 to 983
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Año           984 non-null    int32  
 1   Trimestre     984 non-null    object 
 2   Provincia     984 non-null    object 
 3   ADSL          984 non-null    float64
 4   Cablemodem    984 non-null    float64
 5   Fibra óptica  984 non-null    float64
 6   Wireless      984 non-null    float64
 7   Otros         984 non-null    float64
 8   Total         984 non-null    float64
dtypes: float64(6), int32(1), object(2)
memory usage: 73.0+ KB


In [13]:
accesos_tec['Trimestre'] = accesos_tec['Trimestre'].str.replace('[^0-9]', '').str.rstrip('*')
accesos_tec['Trimestre'] = accesos_tec['Trimestre'].astype(int)

In [14]:
accesos_tec.info()

<class 'pandas.core.frame.DataFrame'>
Index: 984 entries, 0 to 983
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Año           984 non-null    int32  
 1   Trimestre     984 non-null    int32  
 2   Provincia     984 non-null    object 
 3   ADSL          984 non-null    float64
 4   Cablemodem    984 non-null    float64
 5   Fibra óptica  984 non-null    float64
 6   Wireless      984 non-null    float64
 7   Otros         984 non-null    float64
 8   Total         984 non-null    float64
dtypes: float64(6), int32(2), object(1)
memory usage: 69.2+ KB


In [15]:
# Combino el df previo creado con las mismas columnas
df2_comb = pd.merge(accesos_tec, df1_comb, on=['Año', 'Trimestre', 'Provincia'])
df2_comb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 11 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Año                           984 non-null    int32  
 1   Trimestre                     984 non-null    int32  
 2   Provincia                     984 non-null    object 
 3   ADSL                          984 non-null    float64
 4   Cablemodem                    984 non-null    float64
 5   Fibra óptica                  984 non-null    float64
 6   Wireless                      984 non-null    float64
 7   Otros                         984 non-null    float64
 8   Total                         984 non-null    float64
 9   Accesos por cada 100 hab      984 non-null    float64
 10  Accesos por cada 100 hogares  984 non-null    float64
dtypes: float64(8), int32(2), object(1)
memory usage: 77.0+ KB


In [16]:
df2_comb.sample(5)

Año  Trimestre         Provincia      ADSL  Cablemodem  Fibra óptica  \
365  2020          2           Córdoba  282858.0    415007.0       85818.0   
320  2020          4           Formosa   21264.0     10789.0        2432.0   
166  2022          3  Tierra Del Fuego   12754.0     39963.0         403.0   
896  2014          4           Formosa   21245.0      4896.0           1.0   
178  2022          2          La Pampa    7796.0     59273.0       22994.0   

     Wireless   Otros     Total  Accesos por cada 100 hab  \
365   44522.0  8263.0  836468.0                 22.262630   
320   15341.0   380.0   50206.0                  8.268391   
166     125.0   651.0   53896.0                 29.500000   
896       0.0    65.0   26207.0                  4.549163   
178   20200.0  1264.0  111527.0                 30.632722   

     Accesos por cada 100 hogares  
365                     72.713585  
320                     32.487382  
166                     99.510000  
896                     17.956311  
178                     92.980000

In [17]:
# Paso todos los datos a interger
df2_comb = df2_comb.astype({
    'Provincia': 'string',
    'ADSL': 'int',
    'Cablemodem': 'int',
    'Wireless': 'int',
    'Otros': 'int',
    'Total': 'int',
    'Fibra óptica': 'int',
    'Accesos por cada 100 hab': 'int',
    'Accesos por cada 100 hogares': 'int'
})



In [18]:
df2_comb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 11 columns):
 #   Column                        Non-Null Count  Dtype 
---  ------                        --------------  ----- 
 0   Año                           984 non-null    int32 
 1   Trimestre                     984 non-null    int32 
 2   Provincia                     984 non-null    string
 3   ADSL                          984 non-null    int32 
 4   Cablemodem                    984 non-null    int32 
 5   Fibra óptica                  984 non-null    int32 
 6   Wireless                      984 non-null    int32 
 7   Otros                         984 non-null    int32 
 8   Total                         984 non-null    int32 
 9   Accesos por cada 100 hab      984 non-null    int32 
 10  Accesos por cada 100 hogares  984 non-null    int32 
dtypes: int32(10), string(1)
memory usage: 46.3 KB


In [19]:
#Unifico DF con el mismo tipo de informacion.
df3_comb = pd.merge(df2_comb, accesos_baf_pcia, on=['Año', 'Trimestre', 'Provincia'])
df3_comb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 14 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Año                           984 non-null    int32  
 1   Trimestre                     984 non-null    int32  
 2   Provincia                     984 non-null    object 
 3   ADSL                          984 non-null    int32  
 4   Cablemodem                    984 non-null    int32  
 5   Fibra óptica                  984 non-null    int32  
 6   Wireless                      984 non-null    int32  
 7   Otros                         984 non-null    int32  
 8   Total_x                       984 non-null    int32  
 9   Accesos por cada 100 hab      984 non-null    int32  
 10  Accesos por cada 100 hogares  984 non-null    int32  
 11  Banda ancha fija              984 non-null    int64  
 12  Dial up                       982 non-null    float64
 13  Total

---

In [20]:
accesos_totales.sample(3)

Año  Trimestre  Hasta 512 kbps  Entre 512 Kbps y 1 Mbps  \
27  2017          2           22366                   222766   
6   2022          3           33013                    96727   
13  2020          4           39510                    42185   

    Entre 1 Mbps y 6 Mbps  Entre 6 Mbps y 10 Mbps  Entre 10 Mbps y 20 Mbps  \
27                4416065                  990281                  1342000   
6                 1161370                 1151906                   823505   
13                2622638                 1073875                   786595   

    Entre 20 Mbps y 30 Mbps  Más de 30 Mbps   OTROS     Total  
27                   231604           98635       0   7323717  
6                    479822         6995750  349035  11091128  
13                   582420         4239237  185102   9571562

In [21]:
df4_comb = pd.merge(df3_comb, accesos_totales, on=['Año','Trimestre'])


In [22]:
df4_comb.sample(3)

Año  Trimestre Provincia    ADSL  Cablemodem  Fibra óptica  Wireless  \
682  2017          1  La Pampa   12754       47195          5194      6253   
581  2018          1   Córdoba  349342      326894         13178     13237   
758  2016          2   Neuquén   51742       42370          1088     12466   

     Otros  Total_x  Accesos por cada 100 hab  ...  Total_y  Hasta 512 kbps  \
682    517    71913                        20  ...    71913           20653   
581  11275   713926                        19  ...   713926            7309   
758   2849   110515                        17  ...   110515           44008   

     Entre 512 Kbps y 1 Mbps  Entre 1 Mbps y 6 Mbps  Entre 6 Mbps y 10 Mbps  \
682                   230159                4584534                  975027   
581                   215406                3272724                  976956   
758                   384221                5058481                  796998   

     Entre 10 Mbps y 20 Mbps  Entre 20 Mbps y 30 Mbps  Más de 30 Mbps   OTROS  \
682                  1129987                   210668           47990       0   
581                  2068087                   862010          397977  236584   
758                   785759                    73977           29020       0   

       Total  
682  7199018  
581  8037053  
758  7172464  

[3 rows x 23 columns]

---

In [23]:
df_nac = pd.merge(accesos_baf, accesos_vmd, on=['Año', 'Trimestre'])
df_nac.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 41 entries, 0 to 40
Data columns (total 8 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Año                     41 non-null     int64  
 1   Trimestre               41 non-null     int64  
 2   Banda ancha fija        41 non-null     int64  
 3   Dial up                 41 non-null     int64  
 4   Total                   41 non-null     int64  
 5   Periodo                 41 non-null     object 
 6   Mbps (Media de bajada)  41 non-null     float64
 7   Trimestre.1             41 non-null     object 
dtypes: float64(1), int64(5), object(2)
memory usage: 2.7+ KB


In [24]:
df2_nac = pd.merge(df_nac, accesos_vel_prov, on =['Año', 'Trimestre'])
df3_nac = pd.merge(df2_nac, ingresos, on =['Año', 'Trimestre'])
df4_nac = pd.merge(df3_nac, accesos_tec, on =['Año', 'Trimestre'])

In [25]:
df4_nac.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22464 entries, 0 to 22463
Data columns (total 27 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Año                        22464 non-null  int64  
 1   Trimestre                  22464 non-null  int64  
 2   Banda ancha fija           22464 non-null  int64  
 3   Dial up                    22464 non-null  int64  
 4   Total_x                    22464 non-null  int64  
 5   Periodo_x                  22464 non-null  object 
 6   Mbps (Media de bajada)     22464 non-null  float64
 7   Trimestre.1                22464 non-null  object 
 8   Provincia_x                22464 non-null  object 
 9   HASTA 512 kbps             22464 non-null  float64
 10  + 512 Kbps - 1 Mbps        22464 non-null  float64
 11  + 1 Mbps - 6 Mbps          22464 non-null  float64
 12  + 6 Mbps - 10 Mbps         22464 non-null  float64
 13  + 10 Mbps - 20 Mbps        22464 non-null  flo

In [26]:
df_partidos = pd.merge(acceso_tec_loc, mapa_conect, on = ['Provincia', 'Partido', 'Localidad'])
df_partidos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2826 entries, 0 to 2825
Data columns (total 26 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Provincia       2826 non-null   object 
 1   Partido         2826 non-null   object 
 2   Localidad       2826 non-null   object 
 3   Link Indec      2826 non-null   object 
 4   ADSL_x          2826 non-null   int64  
 5   CABLEMODEM      2826 non-null   int64  
 6   DIAL UP         2826 non-null   int64  
 7   FIBRA OPTICA    2826 non-null   int64  
 8   OTROS           2826 non-null   int64  
 9   SATELITAL       2826 non-null   int64  
 10  WIMAX           2826 non-null   int64  
 11  WIRELESS        2826 non-null   int64  
 12  Total general   2826 non-null   int64  
 13  Población       2826 non-null   int64  
 14  ADSL_y          2826 non-null   object 
 15  Cablemódem      2826 non-null   object 
 16  Dial Up         2826 non-null   object 
 17  Fibra óptica    2826 non-null   o

In [27]:
df2_partidos = pd.merge(df_partidos, accesos_loc, on = ['Provincia', 'Partido', 'Localidad'])
df2_partidos.head(5)

Provincia     Partido               Localidad Link Indec_x  ADSL_x  \
0  BUENOS AIRES  25 de Mayo              25 de Mayo      6854100     755   
1  BUENOS AIRES  25 de Mayo               Del Valle      6854020     181   
2  BUENOS AIRES  25 de Mayo       Gobernador Ugarte      6854040       0   
3  BUENOS AIRES  25 de Mayo  Norberto de la Riestra      6854060       0   
4  BUENOS AIRES  25 de Mayo        Lucas Monteverde      6854050       0   

   CABLEMODEM  DIAL UP  FIBRA OPTICA  OTROS  SATELITAL  ...  680 Mbps  \
0        4600        0             2      0        742  ...       NaN   
1           0        0             0     10          1  ...       NaN   
2           0        0             0      0          0  ...       NaN   
3         782        0           496     16          0  ...       NaN   
4           0        0             0      0          0  ...       NaN   

   700 Mbps  800 Mbps  850 Mbps 900 Mbps 999 Mbps 1000 Mbps 1024 Mbps  \
0       NaN       NaN       NaN      NaN      NaN       NaN       NaN   
1       NaN       NaN       NaN      NaN      NaN       NaN       NaN   
2       NaN       NaN       NaN      NaN      NaN       NaN       NaN   
3       NaN       NaN       NaN      NaN      NaN       NaN       NaN   
4       NaN       NaN       NaN      NaN      NaN       NaN       NaN   

  4000 Mbps 10000 Mbps  
0       NaN        NaN  
1       NaN        NaN  
2       NaN        NaN  
3       NaN        NaN  
4       NaN        NaN  

[5 rows x 148 columns]

Duplicados:

In [28]:
# Contar filas duplicadas
numero_duplicados = df2_partidos.duplicated().sum()

print(f"Número de filas duplicadas: {numero_duplicados}")

Número de filas duplicadas: 0


In [29]:
# Elimino las filas duplicadas
df4_nac = df4_nac.drop_duplicates()

# Verifico el número de filas después de eliminar las duplicadas
print(f"El número de filas después de eliminar duplicados: {df4_nac.shape[0]}")


El número de filas después de eliminar duplicados: 22464


---

In [30]:
# Evaluo los resultados del trabajo
igual_1 = (df4_comb['Total'] == df4_comb['Total_x']).sum()
igual_2 = (df4_comb['Total'] == df4_comb['Total_y']).sum()
igual_3 = (df4_comb['Total_x'] == df4_comb['Total_y']).sum()

print(igual_1)
print(igual_2)
print(igual_3)

0
0
984


Visualizo que no existen coincidencias entre 'Total' y 'Total_x', tampoco entre 'Total' y 'Total_y'; pero si hay 984 coincidencias entre 'Total_x' y 'Total_y', se elimina la columna.


In [31]:
df4_comb = df4_comb.drop('Total_y', axis=1)
df4_comb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 22 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Año                           984 non-null    int32  
 1   Trimestre                     984 non-null    int32  
 2   Provincia                     984 non-null    object 
 3   ADSL                          984 non-null    int32  
 4   Cablemodem                    984 non-null    int32  
 5   Fibra óptica                  984 non-null    int32  
 6   Wireless                      984 non-null    int32  
 7   Otros                         984 non-null    int32  
 8   Total_x                       984 non-null    int32  
 9   Accesos por cada 100 hab      984 non-null    int32  
 10  Accesos por cada 100 hogares  984 non-null    int32  
 11  Banda ancha fija              984 non-null    int64  
 12  Dial up                       982 non-null    float64
 13  Hasta

In [32]:
# Renombro algunas columnas necesarias. 
df4_comb = df4_comb.rename(columns = {'Total': 'Total suma Mbps'})
df4_comb = df4_comb.rename(columns = {'Total_x': 'Total conexion'})
df4_comb = df4_comb.rename(columns = {'OTROS': 'Otros Mbps'})
df4_comb = df4_comb.rename(columns = {'HASTA 512 kbps': 'Hasta 512 kbps'})

In [33]:
df4_comb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 22 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Año                           984 non-null    int32  
 1   Trimestre                     984 non-null    int32  
 2   Provincia                     984 non-null    object 
 3   ADSL                          984 non-null    int32  
 4   Cablemodem                    984 non-null    int32  
 5   Fibra óptica                  984 non-null    int32  
 6   Wireless                      984 non-null    int32  
 7   Otros                         984 non-null    int32  
 8   Total conexion                984 non-null    int32  
 9   Accesos por cada 100 hab      984 non-null    int32  
 10  Accesos por cada 100 hogares  984 non-null    int32  
 11  Banda ancha fija              984 non-null    int64  
 12  Dial up                       982 non-null    float64
 13  Hasta

In [34]:
df_group = df4_comb.groupby(['Año', 'Trimestre', 'Provincia']).sum()
df_group.sample(5)

ADSL  Cablemodem  Fibra óptica  Wireless  Otros  \
Año  Trimestre Provincia                                                      
2016 3         Corrientes  55934       30813           217      1354     22   
2021 3         San Luis     8417       11884            27     68856  28680   
2019 4         Río Negro   49747       37933          9658     11437  18365   
2023 3         Mendoza     17677       73066        178719     27609    951   
               San Juan    12669       19376         42383     22317   6985   

                           Total conexion  Accesos por cada 100 hab  \
Año  Trimestre Provincia                                              
2016 3         Corrientes           88340                         8   
2021 3         San Luis            117864                        22   
2019 4         Río Negro           127140                        17   
2023 3         Mendoza             298022                        14   
               San Juan            103730                        12   

                           Accesos por cada 100 hogares  Banda ancha fija  \
Año  Trimestre Provincia                                                    
2016 3         Corrientes                            31             88328   
2021 3         San Luis                              79            117716   
2019 4         Río Negro                             55            126636   
2023 3         Mendoza                               52            297292   
               San Juan                              50            103483   

                           Dial up  Hasta 512 kbps  Entre 512 Kbps y 1 Mbps  \
Año  Trimestre Provincia                                                      
2016 3         Corrientes     12.0           20104                   294413   
2021 3         San Luis      148.0           40174                    41437   
2019 4         Río Negro     504.0           38272                    28980   
2023 3         Mendoza       730.0           30274                    80609   
               San Juan      247.0           30274                    80609   

                           Entre 1 Mbps y 6 Mbps  Entre 6 Mbps y 10 Mbps  \
Año  Trimestre Provincia                                                   
2016 3         Corrientes                4816081                  936533   
2021 3         San Luis                  2550229                 1095772   
2019 4         Río Negro                 2792307                 1046128   
2023 3         Mendoza                    945608                  998725   
               San Juan                   945608                  998725   

                           Entre 10 Mbps y 20 Mbps  Entre 20 Mbps y 30 Mbps  \
Año  Trimestre Provincia                                                      
2016 3         Corrientes                   919208                   124564   
2021 3         San Luis                     710122                   536364   
2019 4         Río Negro                    851619                  1004083   
2023 3         Mendoza                      720084                   350918   
               San Juan                     720084                   350918   

                           Más de 30 Mbps  Otros Mbps  Total suma Mbps  
Año  Trimestre Provincia                                                
2016 3         Corrientes           31022           0          7141925  
2021 3         San Luis           4948174      163269         10085541  
2019 4         Río Negro          2830746      201046          8793181  
2023 3         Mendoza            8003269      277421         11406908  
               San Juan           8003269      277421         11406908

Duplicados:


In [35]:
# Contar filas duplicadas
numero_duplicados = df_group.duplicated().sum()

print(f"Número de filas duplicadas: {numero_duplicados}")

Número de filas duplicadas: 0


-----

In [36]:
df4_nac.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22464 entries, 0 to 22463
Data columns (total 27 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Año                        22464 non-null  int64  
 1   Trimestre                  22464 non-null  int64  
 2   Banda ancha fija           22464 non-null  int64  
 3   Dial up                    22464 non-null  int64  
 4   Total_x                    22464 non-null  int64  
 5   Periodo_x                  22464 non-null  object 
 6   Mbps (Media de bajada)     22464 non-null  float64
 7   Trimestre.1                22464 non-null  object 
 8   Provincia_x                22464 non-null  object 
 9   HASTA 512 kbps             22464 non-null  float64
 10  + 512 Kbps - 1 Mbps        22464 non-null  float64
 11  + 1 Mbps - 6 Mbps          22464 non-null  float64
 12  + 6 Mbps - 10 Mbps         22464 non-null  float64
 13  + 10 Mbps - 20 Mbps        22464 non-null  flo

In [37]:
df4_nac = df4_nac.rename(columns = {'HASTA 512 kbps': 'Hasta 512 kbps'})

In [38]:
df4_nac = df4_nac.drop(['Periodo_y','Periodo_x'], axis = 1)
df4_nac.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22464 entries, 0 to 22463
Data columns (total 25 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Año                        22464 non-null  int64  
 1   Trimestre                  22464 non-null  int64  
 2   Banda ancha fija           22464 non-null  int64  
 3   Dial up                    22464 non-null  int64  
 4   Total_x                    22464 non-null  int64  
 5   Mbps (Media de bajada)     22464 non-null  float64
 6   Trimestre.1                22464 non-null  object 
 7   Provincia_x                22464 non-null  object 
 8   Hasta 512 kbps             22464 non-null  float64
 9   + 512 Kbps - 1 Mbps        22464 non-null  float64
 10  + 1 Mbps - 6 Mbps          22464 non-null  float64
 11  + 6 Mbps - 10 Mbps         22464 non-null  float64
 12  + 10 Mbps - 20 Mbps        22464 non-null  float64
 13  + 20 Mbps - 30 Mbps        22464 non-null  flo

In [39]:
df4_nac

Año  Trimestre  Banda ancha fija  Dial up   Total_x  \
0      2024          1          11621363    11793  11633156   
1      2024          1          11621363    11793  11633156   
2      2024          1          11621363    11793  11633156   
3      2024          1          11621363    11793  11633156   
4      2024          1          11621363    11793  11633156   
...     ...        ...               ...      ...       ...   
22459  2014          1           6362108    36290   6398398   
22460  2014          1           6362108    36290   6398398   
22461  2014          1           6362108    36290   6398398   
22462  2014          1           6362108    36290   6398398   
22463  2014          1           6362108    36290   6398398   

       Mbps (Media de bajada)   Trimestre.1   Provincia_x  Hasta 512 kbps  \
0                  139.150000  Ene-Mar 2024  Buenos Aires         26002.0   
1                  139.150000  Ene-Mar 2024  Buenos Aires         26002.0   
2                  139.150000  Ene-Mar 2024  Buenos Aires         26002.0   
3                  139.150000  Ene-Mar 2024  Buenos Aires         26002.0   
4                  139.150000  Ene-Mar 2024  Buenos Aires         26002.0   
...                       ...           ...           ...             ...   
22459                3.617127  Ene-Mar 2014       Tucumán             6.0   
22460                3.617127  Ene-Mar 2014       Tucumán             6.0   
22461                3.617127  Ene-Mar 2014       Tucumán             6.0   
22462                3.617127  Ene-Mar 2014       Tucumán             6.0   
22463                3.617127  Ene-Mar 2014       Tucumán             6.0   

       + 512 Kbps - 1 Mbps  ...     OTROS    Total_y  \
0                  22510.0  ...  125147.0  5044557.0   
1                  22510.0  ...  125147.0  5044557.0   
2                  22510.0  ...  125147.0  5044557.0   
3                  22510.0  ...  125147.0  5044557.0   
4                  22510.0  ...  125147.0  5044557.0   
...                    ...  ...       ...        ...   
22459              34672.0  ...       0.0   130032.0   
22460              34672.0  ...       0.0   130032.0   
22461              34672.0  ...       0.0   130032.0   
22462              34672.0  ...       0.0   130032.0   
22463              34672.0  ...       0.0   130032.0   

       Ingresos (miles de pesos)          Provincia_y      ADSL  Cablemodem  \
0                   2.804191e+08         Buenos Aires  203297.0   2790663.0   
1                   2.804191e+08      Capital Federal   70108.0   1177475.0   
2                   2.804191e+08            Catamarca    5901.0     11025.0   
3                   2.804191e+08                Chaco   11343.0     58929.0   
4                   2.804191e+08               Chubut   26306.0     77217.0   
...                          ...                  ...       ...         ...   
22459               2.984054e+06           Santa Cruz   18927.0       100.0   
22460               2.984054e+06             Santa Fe  322662.0    174296.0   
22461               2.984054e+06  Santiago Del Estero   32567.0      3598.0   
22462               2.984054e+06     Tierra Del Fuego   21618.0      2837.0   
22463               2.984054e+06              Tucumán  129717.0        83.0   

       Fibra óptica  Wireless     Otros      Total  
0         1799013.0  150290.0  101294.0  5044557.0  
1          203756.0    4493.0   48879.0  1504711.0  
2           56542.0    1384.0      81.0    74933.0  
3           64463.0    8404.0   12404.0   155543.0  
4           26356.0   30867.0    9930.0   170676.0  
...             ...       ...       ...        ...  
22459         381.0    4092.0    3264.0    26764.0  
22460        3059.0    5951.0     644.0   506612.0  
22461          19.0     915.0      34.0    37133.0  
22462         648.0       1.0    2934.0    28038.0  
22463         121.0      13.0      98.0   130032.0  

[22464 rows x 25 columns]

In [40]:
df4_nac.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22464 entries, 0 to 22463
Data columns (total 25 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Año                        22464 non-null  int64  
 1   Trimestre                  22464 non-null  int64  
 2   Banda ancha fija           22464 non-null  int64  
 3   Dial up                    22464 non-null  int64  
 4   Total_x                    22464 non-null  int64  
 5   Mbps (Media de bajada)     22464 non-null  float64
 6   Trimestre.1                22464 non-null  object 
 7   Provincia_x                22464 non-null  object 
 8   Hasta 512 kbps             22464 non-null  float64
 9   + 512 Kbps - 1 Mbps        22464 non-null  float64
 10  + 1 Mbps - 6 Mbps          22464 non-null  float64
 11  + 6 Mbps - 10 Mbps         22464 non-null  float64
 12  + 10 Mbps - 20 Mbps        22464 non-null  float64
 13  + 20 Mbps - 30 Mbps        22464 non-null  flo

In [41]:
#Revision de Iguales
igual_4 = (df4_nac['Total_x'] == df4_nac['Total_y']).sum()
print(igual_4)

0


In [42]:
df4_nac.sample(5)

Año  Trimestre  Banda ancha fija  Dial up   Total_x  \
21958  2014          1           6362108    36290   6398398   
14134  2017          3           7696594    28349   7724943   
11958  2018          3           8320223    26812   8347035   
842    2023          3          11395115    11793  11406908   
21834  2014          2           6428329    36139   6464468   

       Mbps (Media de bajada)    Trimestre.1          Provincia_x  \
21958                3.617127   Ene-Mar 2014            Catamarca   
14134                8.339048  Jul-Sept 2017              Mendoza   
11958               15.360000  Jul-Sept 2018             San Luis   
842                129.670000  Jul-Sept 2023             La Rioja   
21834                3.758712   Abr-Jun 2014  Santiago Del Estero   

       Hasta 512 kbps  + 512 Kbps - 1 Mbps  ...   OTROS   Total_y  \
21958            19.0               9147.0  ...     0.0   24232.0   
14134            32.0                 95.0  ...     0.0  158818.0   
11958            57.0                109.0  ...  1902.0   74362.0   
842              18.0                459.0  ...     0.0   87701.0   
21834          1231.0               9914.0  ...     0.0   37996.0   

       Ingresos (miles de pesos)       Provincia_y     ADSL  Cablemodem  \
21958               2.984054e+06  Tierra Del Fuego  21618.0      2837.0   
14134               1.006600e+07  Tierra Del Fuego  21587.0      7012.0   
11958               1.541098e+07        Corrientes  56444.0     41300.0   
842                 1.331066e+08         Catamarca   7164.0     11066.0   
21834               3.270816e+06          San Luis  11306.0       925.0   

       Fibra óptica  Wireless   Otros    Total  
21958         648.0       1.0  2934.0  28038.0  
14134         622.0       1.0  2244.0  31466.0  
11958         212.0    1468.0    32.0  99456.0  
842         54650.0    1384.0    91.0  74355.0  
21834         354.0      22.0   448.0  13055.0  

[5 rows x 25 columns]

In [43]:
#Elimino los '.'
df4_nac['Hasta 512 kbps'] = df4_nac['Hasta 512 kbps'].astype(str).str.replace('.',',')

In [44]:
# Creo una Funcion para reemplazar los '+' y los '-'
def renombre_col(columna):
    columna = columna.replace('+', 'Entre')
    columna = columna.replace('-','y')
    return columna

In [45]:
df4_nac.rename(columns=renombre_col, inplace=True)

In [46]:
df4_nac.sample(5)

Año  Trimestre  Banda ancha fija  Dial up   Total_x  \
611    2023          3          11395115    11793  11406908   
9687   2019          3           9142891    21793   9164684   
16925  2016          2           7097604    32475   7130079   
5045   2021          3          10075184    10357  10085541   
5069   2021          3          10075184    10357  10085541   

       Mbps (Media de bajada)    Trimestre.1      Provincia_x Hasta 512 kbps  \
611                129.670000  Jul-Sept 2023  Capital Federal          516,0   
9687                24.180000  Jul-Sept 2019       Santa Cruz           28,0   
16925                5.423531   Abr-Jun 2016            Jujuy         2561,0   
5045                48.460000  Jul-Sept 2021         San Luis           60,0   
5069                48.460000  Jul-Sept 2021       Santa Cruz           50,0   

       Entre 512 Kbps y 1 Mbps  ...    OTROS    Total_y  \
611                     4690.0  ...      0.0  1467984.0   
9687                      84.0  ...      0.0    37430.0   
16925                   8292.0  ...      0.0    43496.0   
5045                       0.0  ...  29392.0   117864.0   
5069                       0.0  ...    735.0    48976.0   

       Ingresos (miles de pesos)  Provincia_y      ADSL  Cablemodem  \
611                 1.331066e+08     La Rioja    3726.0     55494.0   
9687                2.165275e+07    Río Negro   67478.0     37425.0   
16925               6.534241e+06      Córdoba  348891.0    277615.0   
5045                4.299994e+07      Córdoba  229275.0    477223.0   
5069                4.299994e+07      Córdoba  229275.0    477223.0   

       Fibra óptica  Wireless    Otros     Total  
611         22054.0    6395.0     32.0   87701.0  
9687         9739.0   10806.0  18847.0  144295.0  
16925        1709.0    5610.0   1089.0  634914.0  
5045       167495.0   57617.0  15178.0  946788.0  
5069       167495.0   57617.0  15178.0  946788.0  

[5 rows x 25 columns]

In [47]:
# Redondeo los numeros 
df4_nac = df4_nac.round(2)

In [48]:
df4_nac.sample(5)

Año  Trimestre  Banda ancha fija  Dial up   Total_x  \
2576   2022          3          11078691    12437  11091128   
5489   2021          2           9852702    10382   9863084   
12649  2018          2           8083533    26911   8110444   
19583  2015          2           6783279    32909   6816188   
17137  2016          2           7097604    32475   7130079   

       Mbps (Media de bajada)    Trimestre.1 Provincia_x Hasta 512 kbps  \
2576                    62.46  Jul-Sept 2022    La Rioja           18,0   
5489                    45.63   Abr-Jun 2021     Mendoza           36,0   
12649                   13.85   Abr-Jun 2018     Tucumán           16,0   
19583                    4.55   Abr-Jun 2015     Tucumán          107,0   
17137                    5.42   Abr-Jun 2016    San Luis           12,0   

       Entre 512 Kbps y 1 Mbps  ...   OTROS   Total_y  \
2576                     689.0  ...     0.0   86864.0   
5489                      31.0  ...    50.0  229671.0   
12649                   6735.0  ...  1305.0  169850.0   
19583                  25178.0  ...     0.0  140670.0   
17137                      4.0  ...     0.0   13950.0   

       Ingresos (miles de pesos)      Provincia_y      ADSL  Cablemodem  \
2576                 64086784.33          Formosa   14470.0     25880.0   
5489                 38239666.63         San Juan   40072.0     12369.0   
12649                14319467.46  Capital Federal  342431.0    977916.0   
19583                 4701790.72          Tucumán  139944.0       479.0   
17137                 6534240.60  Capital Federal  450644.0    892209.0   

       Fibra óptica  Wireless    Otros      Total  
2576        11041.0   16935.0    212.0    68538.0  
5489         6673.0   11372.0   8447.0    78933.0  
12649       35610.0    3908.0   9321.0  1369186.0  
19583         116.0      88.0     50.0   140677.0  
17137       36669.0    2390.0  12672.0  1394584.0  

[5 rows x 25 columns]

In [49]:
#Elimino las columnas redundantes
df4_nac = df4_nac.drop(['Total_y','Total_x'], axis = 1)
df4_nac

Año  Trimestre  Banda ancha fija  Dial up  Mbps (Media de bajada)  \
0      2024          1          11621363    11793                  139.15   
1      2024          1          11621363    11793                  139.15   
2      2024          1          11621363    11793                  139.15   
3      2024          1          11621363    11793                  139.15   
4      2024          1          11621363    11793                  139.15   
...     ...        ...               ...      ...                     ...   
22459  2014          1           6362108    36290                    3.62   
22460  2014          1           6362108    36290                    3.62   
22461  2014          1           6362108    36290                    3.62   
22462  2014          1           6362108    36290                    3.62   
22463  2014          1           6362108    36290                    3.62   

        Trimestre.1   Provincia_x Hasta 512 kbps  Entre 512 Kbps y 1 Mbps  \
0      Ene-Mar 2024  Buenos Aires        26002,0                  22510.0   
1      Ene-Mar 2024  Buenos Aires        26002,0                  22510.0   
2      Ene-Mar 2024  Buenos Aires        26002,0                  22510.0   
3      Ene-Mar 2024  Buenos Aires        26002,0                  22510.0   
4      Ene-Mar 2024  Buenos Aires        26002,0                  22510.0   
...             ...           ...            ...                      ...   
22459  Ene-Mar 2014       Tucumán            6,0                  34672.0   
22460  Ene-Mar 2014       Tucumán            6,0                  34672.0   
22461  Ene-Mar 2014       Tucumán            6,0                  34672.0   
22462  Ene-Mar 2014       Tucumán            6,0                  34672.0   
22463  Ene-Mar 2014       Tucumán            6,0                  34672.0   

       Entre 1 Mbps y 6 Mbps  ...  Entre 30 Mbps     OTROS  \
0                   221185.0  ...      4068292.0  125147.0   
1                   221185.0  ...      4068292.0  125147.0   
2                   221185.0  ...      4068292.0  125147.0   
3                   221185.0  ...      4068292.0  125147.0   
4                   221185.0  ...      4068292.0  125147.0   
...                      ...  ...            ...       ...   
22459                83210.0  ...            0.0       0.0   
22460                83210.0  ...            0.0       0.0   
22461                83210.0  ...            0.0       0.0   
22462                83210.0  ...            0.0       0.0   
22463                83210.0  ...            0.0       0.0   

       Ingresos (miles de pesos)          Provincia_y      ADSL  Cablemodem  \
0                   2.804191e+08         Buenos Aires  203297.0   2790663.0   
1                   2.804191e+08      Capital Federal   70108.0   1177475.0   
2                   2.804191e+08            Catamarca    5901.0     11025.0   
3                   2.804191e+08                Chaco   11343.0     58929.0   
4                   2.804191e+08               Chubut   26306.0     77217.0   
...                          ...                  ...       ...         ...   
22459               2.984054e+06           Santa Cruz   18927.0       100.0   
22460               2.984054e+06             Santa Fe  322662.0    174296.0   
22461               2.984054e+06  Santiago Del Estero   32567.0      3598.0   
22462               2.984054e+06     Tierra Del Fuego   21618.0      2837.0   
22463               2.984054e+06              Tucumán  129717.0        83.0   

      Fibra óptica  Wireless     Otros      Total  
0        1799013.0  150290.0  101294.0  5044557.0  
1         203756.0    4493.0   48879.0  1504711.0  
2          56542.0    1384.0      81.0    74933.0  
3          64463.0    8404.0   12404.0   155543.0  
4          26356.0   30867.0    9930.0   170676.0  
...            ...       ...       ...        ...  
22459        381.0    4092.0    3264.0    26764.0  
22460       3059.0    5951.0     644.0   506612.

In [50]:
#Renombramos para un mejor entendimiento.
df4_nac = df4_nac.rename(columns = {'OTROS': 'Otros Mbps'})
df4_nac = df4_nac.rename(columns = {'Otros': 'Otras conexiones'})
df4_nac = df4_nac.rename(columns = {'Total': 'Total Mbps-conexiones'})
df4_nac = df4_nac.rename(columns = {'Entre 30 Mbps': 'Mas de 30 Mbps'})
df4_nac = df4_nac.rename(columns = {'Provincia_y': 'Provincia'})

In [51]:
df4_nac.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22464 entries, 0 to 22463
Data columns (total 23 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Año                        22464 non-null  int64  
 1   Trimestre                  22464 non-null  int64  
 2   Banda ancha fija           22464 non-null  int64  
 3   Dial up                    22464 non-null  int64  
 4   Mbps (Media de bajada)     22464 non-null  float64
 5   Trimestre.1                22464 non-null  object 
 6   Provincia_x                22464 non-null  object 
 7   Hasta 512 kbps             22464 non-null  object 
 8   Entre 512 Kbps y 1 Mbps    22464 non-null  float64
 9   Entre 1 Mbps y 6 Mbps      22464 non-null  float64
 10  Entre 6 Mbps y 10 Mbps     22464 non-null  float64
 11  Entre 10 Mbps y 20 Mbps    22464 non-null  float64
 12  Entre 20 Mbps y 30 Mbps    22464 non-null  float64
 13  Mas de 30 Mbps             22464 non-null  flo

In [52]:
columns_to_convert = [
    'Ingresos (miles de pesos)', 
    'ADSL', 
    'Cablemodem', 
    'Fibra óptica', 
    'Wireless', 
    'Otras conexiones', 
    'Total Mbps-conexiones'
]

df4_nac[columns_to_convert] = df4_nac[columns_to_convert].astype(int)

In [53]:
# cambio los'.' por ','
df4_nac['Hasta 512 kbps'] = df4_nac['Hasta 512 kbps'].str.replace(',', '.')

In [54]:
print(df4_nac['Entre 512 Kbps y 1 Mbps'].dtype)


float64


In [55]:
df4_nac['Entre 512 Kbps y 1 Mbps'] = df4_nac['Entre 512 Kbps y 1 Mbps'].astype(str)


In [56]:
df4_nac['Entre 512 Kbps y 1 Mbps'] = df4_nac['Entre 512 Kbps y 1 Mbps'].str.replace(',', '.')

In [57]:
a_float = ['Hasta 512 kbps',
    'Entre 512 Kbps y 1 Mbps']
df4_nac[a_float] = df4_nac[a_float].astype(float)

In [58]:
df4_nac.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22464 entries, 0 to 22463
Data columns (total 23 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Año                        22464 non-null  int64  
 1   Trimestre                  22464 non-null  int64  
 2   Banda ancha fija           22464 non-null  int64  
 3   Dial up                    22464 non-null  int64  
 4   Mbps (Media de bajada)     22464 non-null  float64
 5   Trimestre.1                22464 non-null  object 
 6   Provincia_x                22464 non-null  object 
 7   Hasta 512 kbps             22464 non-null  float64
 8   Entre 512 Kbps y 1 Mbps    22464 non-null  float64
 9   Entre 1 Mbps y 6 Mbps      22464 non-null  float64
 10  Entre 6 Mbps y 10 Mbps     22464 non-null  float64
 11  Entre 10 Mbps y 20 Mbps    22464 non-null  float64
 12  Entre 20 Mbps y 30 Mbps    22464 non-null  float64
 13  Mas de 30 Mbps             22464 non-null  flo

Duplicados:

In [59]:
# Contar filas duplicadas
numero_duplicados = df4_nac.duplicated().sum()

print(f"Número de filas duplicadas: {numero_duplicados}")


Número de filas duplicadas: 0


-----

In [60]:
df2_partidos.info(3)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2804 entries, 0 to 2803
Data columns (total 148 columns):
 #    Column          Dtype  
---   ------          -----  
 0    Provincia       object 
 1    Partido         object 
 2    Localidad       object 
 3    Link Indec_x    object 
 4    ADSL_x          int64  
 5    CABLEMODEM      int64  
 6    DIAL UP         int64  
 7    FIBRA OPTICA    int64  
 8    OTROS           int64  
 9    SATELITAL       int64  
 10   WIMAX           int64  
 11   WIRELESS        int64  
 12   Total general   int64  
 13   Población       int64  
 14   ADSL_y          object 
 15   Cablemódem      object 
 16   Dial Up         object 
 17   Fibra óptica    object 
 18   Satelital       object 
 19   Wireless        object 
 20   Telefonía Fija  object 
 21   3G              object 
 22   4G              object 
 23   Link            float64
 24   Latitud         float64
 25   Longitud        float64
 26   Link Indec_y    object 
 27   Otros           

In [61]:
df2_partidos = df2_partidos.merge(mapa_conect[['Partido', 'Latitud', 'Longitud']], on='Partido', how='left')


In [62]:
df2_partidos.sample(5)

Provincia                        Partido         Localidad  \
37797     RIO NEGRO                   General Roca  Villa San Isidro   
10679         CHACO  Libertador General San Martín     Pampa Almirón   
8228   BUENOS AIRES                   Tres Arroyos            Orense   
40538      SAN JUAN                         Rawson       El Medanito   
28907         JUJUY                        Susques      Olaroz Chico   

      Link Indec_x  ADSL_x  CABLEMODEM  DIAL UP  FIBRA OPTICA  OTROS  \
37797     62042570       0           0        0            44      0   
10679     22084050       0           0        0            26      0   
8228       6833060       0           0        1           246    407   
40538     70077010       0           0        0             0      0   
28907     38084060       0           0        0            30      0   

       SATELITAL  ...  800 Mbps  850 Mbps  900 Mbps  999 Mbps 1000 Mbps  \
37797          0  ...       NaN       NaN       NaN       NaN       NaN   
10679          0  ...       NaN       NaN       NaN       NaN       NaN   
8228           0  ...       NaN       NaN       NaN       NaN       NaN   
40538          0  ...       NaN       NaN       NaN       NaN       NaN   
28907          0  ...       NaN       NaN       NaN       NaN       NaN   

      1024 Mbps 4000 Mbps 10000 Mbps  Latitud_y Longitud_y  
37797       NaN       NaN        NaN -39.043600 -67.554400  
10679       NaN       NaN        NaN -26.702248 -59.123785  
8228        NaN       NaN        NaN -38.492679 -60.468423  
40538       NaN       NaN        NaN -31.569619 -68.557769  
28907       NaN       NaN        NaN -23.259929 -67.023689  

[5 rows x 150 columns]

In [63]:
df2_partidos.dtypes

Provincia        object
Partido          object
Localidad        object
Link Indec_x     object
ADSL_x            int64
                 ...   
1024 Mbps       float64
4000 Mbps       float64
10000 Mbps      float64
Latitud_y       float64
Longitud_y      float64
Length: 150, dtype: object

In [64]:
# Renombro Columnas
nuevo_nombre = {'ADSL_x': 'ADSL',
                'FIBRA OPTICA': 'Fibra Optica',
                'OTROS': 'Otras Conexiones',
                'CABLEMODEM': 'Cable Modem',
                'SATELITAL': 'Satelital',
                'Telefonía Fija': 'Telefonia Fija',
                'Latitud_y': 'Latitud',
                'Longitud_y': 'Longitud'}

In [65]:
df2_partidos = df2_partidos.rename(columns = nuevo_nombre)

In [66]:
#Visualizo
df2_partidos.head(5)

Provincia     Partido   Localidad Link Indec_x  ADSL  Cable Modem  \
0  BUENOS AIRES  25 de Mayo  25 de Mayo      6854100   755         4600   
1  BUENOS AIRES  25 de Mayo  25 de Mayo      6854100   755         4600   
2  BUENOS AIRES  25 de Mayo  25 de Mayo      6854100   755         4600   
3  BUENOS AIRES  25 de Mayo  25 de Mayo      6854100   755         4600   
4  BUENOS AIRES  25 de Mayo  25 de Mayo      6854100   755         4600   

   DIAL UP  Fibra Optica  Otras Conexiones  Satelital  ...  800 Mbps  \
0        0             2                 0        742  ...       NaN   
1        0             2                 0        742  ...       NaN   
2        0             2                 0        742  ...       NaN   
3        0             2                 0        742  ...       NaN   
4        0             2                 0        742  ...       NaN   

   850 Mbps  900 Mbps  999 Mbps 1000 Mbps 1024 Mbps 4000 Mbps 10000 Mbps  \
0       NaN       NaN       NaN       NaN       NaN       NaN        NaN   
1       NaN       NaN       NaN       NaN       NaN       NaN        NaN   
2       NaN       NaN       NaN       NaN       NaN       NaN        NaN   
3       NaN       NaN       NaN       NaN       NaN       NaN        NaN   
4       NaN       NaN       NaN       NaN       NaN       NaN        NaN   

     Latitud   Longitud  
0 -35.433939 -60.173121  
1 -35.739217 -60.559063  
2 -35.897345 -60.731609  
3 -35.270234 -59.557503  
4 -35.164397 -60.081358  

[5 rows x 150 columns]

Duplicados:

In [72]:
# Contar filas duplicadas
numero_duplicados = df2_partidos.duplicated().sum()

print(f"Número de filas duplicadas: {numero_duplicados}")

Número de filas duplicadas: 600


In [73]:
# Elimino filas duplicadas
df4_nac = df4_nac.drop_duplicates()

# Verifico el número de filas después de eliminar duplicados
print(f"Número de filas después de eliminar duplicados: {df4_nac.shape[0]}")


Número de filas después de eliminar duplicados: 22464


----

Visualizo los DataFrames resultantes

In [74]:
df4_comb.sample(2)

Año  Trimestre Provincia    ADSL  Cablemodem  Fibra óptica  Wireless  \
585  2018          1     Jujuy   33856       10456            23        28   
29   2023          4   Córdoba  115662      452380        440536     61862   

     Otros  Total conexion  Accesos por cada 100 hab  ...  Dial up  \
585     43           44406                         5  ...      1.0   
29    8862         1079302                        27  ...     63.0   

     Hasta 512 kbps  Entre 512 Kbps y 1 Mbps  Entre 1 Mbps y 6 Mbps  \
585            7309                   215406                3272724   
29            29708                    71742                 900253   

     Entre 6 Mbps y 10 Mbps  Entre 10 Mbps y 20 Mbps  Entre 20 Mbps y 30 Mbps  \
585                  976956                  2068087                   862010   
29                   978108                   697232                   350290   

     Más de 30 Mbps  Otros Mbps  Total suma Mbps  
585          397977      236584          8037053  
29          8224736      295592         11547661  

[2 rows x 22 columns]

In [75]:
df_group.sample(2)

ADSL  Cablemodem  Fibra óptica  Wireless  Otros  \
Año  Trimestre Provincia                                                     
2022 2         Catamarca  11928        9430         42087      1329   2234   
2019 2         Chubut     37313       26338          1917     14443   7545   

                          Total conexion  Accesos por cada 100 hab  \
Año  Trimestre Provincia                                             
2022 2         Catamarca           67008                        15   
2019 2         Chubut              87556                        14   

                          Accesos por cada 100 hogares  Banda ancha fija  \
Año  Trimestre Provincia                                                   
2022 2         Catamarca                            62             67007   
2019 2         Chubut                               46             86038   

                          Dial up  Hasta 512 kbps  Entre 512 Kbps y 1 Mbps  \
Año  Trimestre Provincia                                                     
2022 2         Catamarca      1.0           33667                    99498   
2019 2         Chubut      1518.0          202790                    95980   

                          Entre 1 Mbps y 6 Mbps  Entre 6 Mbps y 10 Mbps  \
Año  Trimestre Provincia                                                  
2022 2         Catamarca                1193090                 1197030   
2019 2         Chubut                   3444458                  871142   

                          Entre 10 Mbps y 20 Mbps  Entre 20 Mbps y 30 Mbps  \
Año  Trimestre Provincia                                                     
2022 2         Catamarca                   856562                   485321   
2019 2         Chubut                     1169632                  1250366   

                          Más de 30 Mbps  Otros Mbps  Total suma Mbps  
Año  Trimestre Provincia                                               
2022 2         Catamarca         6741922      351594         10958684  
2019 2         Chubut            1677830      247983          8960181

In [76]:
df_nac.sample(2)

Año  Trimestre  Banda ancha fija  Dial up     Total       Periodo  \
5   2022          4          11195668    12446  11208114  Oct-Dic 2022   
28  2017          1           7277314    28540   7305854  Ene-Mar 2017   

    Mbps (Media de bajada)   Trimestre.1  
5               106.560000  Oct-Dic 2022  
28                6.624079  Ene-Mar 2017

In [77]:
df2_partidos.sample(2)

Provincia      Partido                        Localidad Link Indec_x  \
12363   CORDOBA  Calamuchita  Villa Ciudad Parque Los Reartes     14007170   
41957  SANTA FE  Castellanos                    Bauer y Sigel     82021060   

       ADSL  Cable Modem  DIAL UP  Fibra Optica  Otras Conexiones  Satelital  \
12363     0          497        0             0                 0          0   
41957     0            0        0             0                 0          0   

       ...  800 Mbps  850 Mbps  900 Mbps  999 Mbps 1000 Mbps 1024 Mbps  \
12363  ...       NaN       NaN       NaN       NaN       NaN       NaN   
41957  ...       NaN       NaN       NaN       NaN       NaN       NaN   

      4000 Mbps 10000 Mbps    Latitud   Longitud  
12363       NaN        NaN -32.103783 -64.754103  
41957       NaN        NaN -31.552633 -61.546265  

[2 rows x 150 columns]

----

Datasets finales luego del ETL:

In [79]:
#  Guardo los DataFrames resultantes
df4_comb.to_csv('df_combinado.csv',)
df_group.to_csv('df_agrupado.csv')
df4_nac.to_csv('df_nac.csv')
df2_partidos.to_csv('df_partido.csv')                                                                                                    

---

Visualizo el 5to Dataset de interes:


In [81]:
df_prov = pd.read_csv('./df_provincias.csv')
df_prov.sample(2)

centroide_lat  centroide_lon  id  nombre       nombre_completo
6      -43.788623     -68.526759  26  Chubut  Provincia del Chubut
20     -23.320078     -65.764252  38   Jujuy    Provincia de Jujuy

Duplicados:

In [82]:
# Contar filas duplicadas
numero_duplicados = df_prov.duplicated().sum()

print(f"Número de filas duplicadas: {numero_duplicados}")

Número de filas duplicadas: 0


In [83]:
df_prov.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 24 entries, 0 to 23
Data columns (total 5 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   centroide_lat    24 non-null     float64
 1   centroide_lon    24 non-null     float64
 2   id               24 non-null     int64  
 3   nombre           24 non-null     object 
 4   nombre_completo  24 non-null     object 
dtypes: float64(2), int64(1), object(2)
memory usage: 1.1+ KB
